### PR TITLE
Fix unnecessary float to int cast for strtol

### LINKED ISSUE #732: AS that does not use floats should not compile into Wasm that uses floats
Floats cannot be used in Wasm contracts that are running on blockchains due to the non-determinism. The way to prevent it is to disallow Wasm that has floats, by inspecting it before execution. Unfortunately, in some cases AssemblyScript that does not use floats can compile to Wasm that uses floats, e.g. when formatting strings. Our [blockchain](https://nearprotocol.com) uses both AssemblyScript and Rust for smart contracts and floats are causing significant issues for us because AssemblyScript prevents us from disabling floats. I think it is a high priority for AssemblyScript to fix it, since the blockchains constitute a very large user-base of AssemblyScript and Wasm.

### PR DESCRIPTION
**Pull request by MaxGraey**
Fix #732

### CODE REVIEW

**Comment by MaxGraey** [comment]
Old behaviour cause to "invalid integer conversion" trap 
```wat
  f64.const nan:0x8000000000000
  i32.trunc_f64_s
```
So may be better throw exception instead return zero for non-float values.
@dcodeIO wdyt?

**Comment by dcodeIO** [comment]
Looks good. I believe the sole reason for the `NaN`s there was compatibility of return values when used inside `parseInt`, which operates on `f64` and returns `NaN` on invalid inputs etc. Previous behavior for `i32` was returning `0` (after casting), and since that remains the same (but removes the casting), the fix should be just fine.

File: std/assembly/util/string.ts [hunk 1]
```diff
@@ -85,13 +88,19 @@ export function strtol<T>(str: string, radix: i32 = 0): T {
   }
   if (code == CharCode.MINUS) {
     // @ts-ignore: cast
-    if (!--len) return <T>NaN;
+    if (!--len) {
+      if (isFloat<T>()) return <T>NaN;
```
Comment (dcodeIO) [inline std/assembly/util/string.ts]: Does the ts-ignore above still catch this? Might be that it now requires to move the ignore to this line and add one to the next.

Comment (MaxGraey) [reply std/assembly/util/string.ts]: yeah. Fixed

**Pull request merged**